### ISSUE TITLE
[promise-function-async] False positives for inline arrow funcs

### ISSUE THREAD

**Issue by milesj** [typescript-eslint/typescript-eslint#778]
Noticed a lot of false positives for inline arrow functions that were returning non-promise values, yet were failing this rule. `touchedFiles` is a `string[]`.

<img width="933" alt="Screen Shot 2019-07-29 at 12 56 53 PM" src="https://user-images.githubusercontent.com/143744/62078409-78a06600-b201-11e9-97bc-27a846e2b206.png">
<img width="974" alt="Screen Shot 2019-07-29 at 12 57 15 PM" src="https://user-images.githubusercontent.com/143744/62078410-78a06600-b201-11e9-8eba-9f1d27e728e6.png">
<img width="933" alt="Screen Shot 2019-07-29 at 1 05 30 PM" src="https://user-images.githubusercontent.com/143744/62078488-98378e80-b201-11e9-87f6-58f4b8bac474.png">

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/promise-function-async": "warn"
  }
}
```

```TS
import path from 'path';

export function someExample() {
  ['foo', 'bar', 'baz'].some(value => path.basename(value));
}
```

**Expected Result**

Inline arrow functions that accept/return primitive values should not fail.

**Actual Result**

They currently fail.

**Additional Info**

Doesn't fail when `checkArrowFunctions` is false.

TS config:
```
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "declaration": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "jsx": "preserve",
    "lib": [
      "dom",
      "esnext"
    ],
    "module": "commonjs",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "pretty": true,
    "removeComments": false,
    "strict": true,
    "target": "es2018",
    "declarationDir": "./lib",
    "outDir": "./lib"
  },
  "include": [
    "./src/**/*",
    "./types/**/*"
  ],
  "exclude": [
    "**/node_modules/*"
  ]
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.14.0` |
| `npm`                              | `6.5.0` |

**Comment by bradzacher** [comment]
try setting `allowAny = true`.
This will be the default setting in the 2.0.0 release

**Comment by milesj** [comment]
@bradzacher That works but none of these arrow funcs return `any`, so it seems odd/confusing.

**Comment by bradzacher** [comment]
yeah looking at your examples it shouldn't be reporting for them.
I'll leave this open, but I don't know if/when we'll work on it considering that in 2.0.0 we changed the default for the setting. We were going to remove it but I left it in for legacy reasons.

Something's funny with the `any` detection in the rule.

**Comment by bradzacher** [comment]
This is no longer reproing in master.
I think this was fixed in #1023
